File: mo-system/mo-system-service/src/main/java/ink/ckx/mo/system/mapper/SysRoleMenuMapper.java
package ink.ckx.mo.system.mapper;

import com.baomidou.mybatisplus.core.mapper.BaseMapper;
import ink.ckx.mo.system.api.model.entity.SysRoleMenu;
import org.apache.ibatis.annotations.Mapper;

import java.util.List;

/**
 * 角色<->菜单持久层
 *
 * @author chenkaixin
 */
@Mapper
public interface SysRoleMenuMapper extends BaseMapper<SysRoleMenu> {

    /**
     * 获取角色拥有的菜单ID集合
     *
     * @param roleId
     * @return
     */
    List<Long> listMenuIdsByRoleId(Long roleId);
}

File: mo-admin/mo-admin-service/src/main/kotlin/ink/ckx/mo/admin/mapper/SysRoleMenuMapper.kt
package ink.ckx.mo.admin.mapper

import com.baomidou.mybatisplus.core.mapper.BaseMapper
import ink.ckx.mo.admin.api.model.entity.SysRoleMenu
import org.apache.ibatis.annotations.Mapper

/**
 * 角色<->菜单持久层
 *
 * @author chenkaixin
 */
@Mapper
interface SysRoleMenuMapper : BaseMapper<SysRoleMenu> {

    /**
     * 获取角色拥有的菜单ID集合
     *
     * @param roleId
     * @return
     */
    fun listMenuIdsByRoleId(roleId: Long): List<Long>
}